src/codecoverage/bot/code_coverage_bot/codecov.py

```diff
@@ -116,12 +116,27 @@ def retrieve_source_and_artifacts(self):
             # Thread 2 - Clone repository.
             executor.submit(self.clone_repository, self.repository, self.revision)
 
+    def generate_covdir(self):
+        '''
+        Build the covdir report using current artifacts
+        '''
+        output = grcov.report(
+            self.artifactsHandler.get(),
+            source_dir=self.repo_dir,
+            out_format='covdir',
+        )
+        logger.info('Covdir report generated successfully')
+        return output
+
     # This function is executed when the bot is triggered at the end of a mozilla-central build.
     def go_from_trigger_mozilla_central(self):
         commit_sha = self.githubUtils.mercurial_to_git(self.revision)
         try:
             uploader.get_codecov(commit_sha)
             logger.warn('Build was already injested')
+
+            # Update GCP covdir report anyway
+            uploader.gcp(self.branch, self.revision, self.generate_covdir())
```

Comment:
This won't work without the call to retrieve_source_and_artifacts which is a bit later.
I would also check, before doing this, if the report is not already there in gcp.

Reply:
Done !